### PR TITLE
docs: move to gridsome

### PR DESCRIPTION
**Pull request by andrewrynhard**
Brings in a new theme, improved content, and restructured layout.

Signed-off-by: Andrew Rynhard <andrew@rynhard.io>

### CODE REVIEW

**Review by smira — APPROVED** [review]
Make it live! We can fix small nits later!

File: website/content/docs/v0.7/Guides/configuring-the-cluster-endpoint.md [hunk 1]
```diff
@@ -30,9 +31,9 @@ The configuration can either be done on a Loadbalancer, or simply trough DNS.
 For example:
 
 > This is in the config file for the cluster e.g. init.yaml, controlplane.yaml and join.yaml.
-> for more details, please see: [v1alpha1 endpoint configuration](https://www.talos.dev/docs/v0.7/en/configuration/v1alpha1#controlplane)
+> for more details, please see: [v1alpha1 endpoint configuration](https://www.talos.dev/docs/v0.6/en/configuration/v1alpha1#controlplane)
```
Comment (Unix4ever) [inline website/content/docs/v0.7/Guides/configuring-the-cluster-endpoint.md:34]: is 0.6 version in 0.7 configs intended?
There are many other places where 0.6 is referenced from 0.7.

Comment (andrewrynhard) [reply website/content/docs/v0.7/Guides/configuring-the-cluster-endpoint.md:34]: whoops, good catch :+1:

Comment (andrewrynhard) [reply website/content/docs/v0.7/Guides/configuring-the-cluster-endpoint.md]: Fixed.

File: website/content/docs/v0.7/Guides/configuring-pull-through-cache.md [hunk 2]
```diff
@@ -14,11 +14,11 @@ It can be also used to verify that all the images are available in local registr
 The follow are requirements for creating the set of caching proxies:
 
 - Docker 18.03 or greater
-- Local cluster requirements for either [docker](docker) or [QEMU](qemu).
+- Local cluster requirements for either [docker](docker) or [fireracker](firecracker).
```
Comment (Unix4ever) [inline website/content/docs/v0.7/Guides/configuring-pull-through-cache.md:17]: this file looks to be reverted. Firecracker and no ghcr io.

File: website/content/docs/v0.7/Bare Metal Platforms/digital-rebar.md [hunk 3]
```diff
@@ -80,7 +80,7 @@ We have some Digital Rebar [example files](https://github.com/talos-systems/talo
 To apply these configs you need to create them, and then apply them as follow:
 
 ```bash
-$ drpcli bootenvs create talos-0.7
+$ drpcli bootenvs create talos-0.6
```
Comment (Unix4ever) [inline website/content/docs/v0.7/Bare Metal Platforms/digital-rebar.md:83]: probably here we have more unintentional 0.6 references.

**Review by Unix4ever — APPROVED** [review]
lgtm with couple more small comments.

File: website/content/docs/v0.7/Bare Metal Platforms/digital-rebar.md (hunk 3 above)
Comment (andrewrynhard) [reply website/content/docs/v0.7/Bare Metal Platforms/digital-rebar.md:83]: :+1: Thank you.

File: website/content/docs/v0.7/Guides/configuring-pull-through-cache.md (hunk 2 above)
Comment (andrewrynhard) [reply website/content/docs/v0.7/Guides/configuring-pull-through-cache.md:17]: :+1:

**Pull request merged**